js/data.js

```diff
@@ -1 +1,14 @@
 /* exported data */
+
+var data = {
+  favorites: []
+};
+
+window.addEventListener('beforeunload', function () {
+  var favoriteString = JSON.stringify(data.favorites);
+  this.localStorage.setItem('favoriteAgents', favoriteString);
+});
+
+if (localStorage.getItem('favorites')) {
+  data.favorites = JSON.parse(localStorage.getItem('favorites'));
+}
```

Comment:
You need to be looking for 'favoriteAgents'. You're saving the key as 'favoriteAgents', so 'favorites' doesn't exist.